core/domain/html_validation_service.py

```diff
@@ -880,6 +880,37 @@ def get_filename_with_dimensions(old_filename, exp_id):
     return new_filename
 
 
+def add_svg_filename_to_math_rte_components(html_string):
+    """Replaces the attribute raw_latex-with-value in all Math RTE tags with
+    a new attribute math_content-with-value. The new attribute has an additional
+    field for storing SVG filenames.
+
+    Args:
+        html_string: str. HTML string to modify.
+
+    Returns:
+        str. Updated HTML string with all Math RTE tags having the new
+        attribute.
+    """
+
+    soup = bs4.BeautifulSoup(
+        html_string.encode(encoding='utf-8'), 'html.parser')
+    for math in soup.findAll(name='oppia-noninteractive-math'):
+        if math.has_attr('raw_latex-with-value'):
+            raw_latex = json.loads(unescape_html(math['raw_latex-with-value']))
+            # Generate an SVG file and filename from the given Latex string.
+            math_expression_contents_dict = {
+                'raw_latex': raw_latex,
+                'svg_filename': ''
+            }
+            # Delete the attribute raw_latex-with-value.
+            del math['raw_latex-with-value']
+            # Add the new attribute math_expression_contents-with-value.
+            math['math_content-with-value'] = escape_html(
+                json.dumps(math_expression_contents_dict, sort_keys=True))
+    return python_utils.UNICODE(soup).replace('<br/>', '<br>')
```

Comment:
`replace('<br/>', '<br>')`, why this?

Reply:
This is used in the other functions in `html_validation_service.py` too. Beautiful soup replaces <br> with </br> tag, but for passing the text angular migration we should have only <br> tags and no </br> .
Hence we replace this.

Reply:
Can you add this comment as "code comment", as a reader I was confused!